### PR TITLE
replace stdb::vector with stdb_vector

### PR DESCRIPTION
**Pull request by lovio**

### CODE REVIEW

File: proto/fbe.h [hunk 1]
```diff
@@ -171,20 +176,20 @@ class buffer_t
...
     void assign(size_t size, uint8_t value) { _data.assign(size, value); }
     void assign(const uint8_t* data, size_t size) { _data.assign(data, data + size); }
     template <class InputIterator>
     void assign(InputIterator first, InputIterator last) { _data.assign(first, last); }
-    iterator insert(const_iterator position, uint8_t value) { return _data.insert(position, value); }
-    iterator insert(const_iterator position, const std::string& str) { return insert(position, (const uint8_t*)str.c_str(), str.size()); }
-    iterator insert(const_iterator position, const std::vector<uint8_t>& vec) { return insert(position, vec.begin(), vec.end()); }
-    iterator insert(const_iterator position, size_t size, uint8_t value) { return _data.insert(position, size, value); }
-    iterator insert(const_iterator position, const uint8_t* data, size_t size) { return _data.insert(position, data, data + size); }
+    iterator insert(iterator position, uint8_t value) { return _data.insert(position, value); }
```
Comment (hurricane1026) [inline proto/fbe.h:184]: 是不是应该保留 const_iterator 啊？

File: proto/extra_ptr.cpp [hunk 2]
```diff
@@ -153,8 +155,9 @@ Extra::Extra(int64_t arg_num, const stdb::memory::string& arg_data, std::unique_
     , infol(std::move(arg_infol))
     , infopl()
 {
+    infopv.reserve(arg_infopv.size());
     for (auto& it : arg_infopv)
-        infopv.emplace_back(it.release());
+        infopv.emplace_back<Safety::Unsafe>(it.release());
     for (auto& it : arg_infopl)
         infopl.emplace_back(it.release());
```
Comment (hurricane1026) [inline proto/extra_ptr.cpp:162]: 这里不能 unsafe 么？

File: proto/fbe.h (hunk 1 above)
Comment (lovio) [reply proto/fbe.h:184]: stdb_vector得支持才行

File: proto/extra_ptr.cpp [hunk 3]
```diff
@@ -153,8 +155,9 @@ Extra::Extra(int64_t arg_num, const stdb::memory::string& arg_data, std::unique_
     , infol(std::move(arg_infol))
     , infopl()
 {
+    infopv.reserve(arg_infopv.size());
     for (auto& it : arg_infopv)
-        infopv.emplace_back(it.release());
+        infopv.emplace_back<Safety::Unsafe>(it.release());
```
Comment (hurricane1026) [inline proto/extra_ptr.cpp:160]: 哪里 reserve 了？不 reserve 就调用 unsafe 应该不 safe

File: source/generator_cpp.inl [hunk 4]
```diff
@@ -565,13 +565,13 @@ inline void FieldModelCustomArray<T, TStruct, N>::get(std::vector<TStruct>& valu
     {
         TStruct value;
         fbe_model.get(value);
-        values.emplace_back(std::move(value));
+        values.template emplace_back<Safety::Unsafe>(std::move(value));
```
Comment (hurricane1026) [inline source/generator_cpp.inl:568]: reserve 代码在哪？

File: proto/extra_ptr.cpp (hunk 3 above)
Comment (lovio) [reply proto/extra_ptr.cpp:160]: https://github.com/stdbio/FBE/pull/25/files/ca7aff82b3e13a1cc534ab5137229d633c7a9e01#diff-28426c8d92051edba7cffb914c3e0d548919cca94f79a49b8ceba595a292b07dR158

File: proto/extra_ptr.cpp (hunk 2 above)
Comment (lovio) [reply proto/extra_ptr.cpp:162]: 这个不是vector

File: source/generator_cpp.inl (hunk 4 above)
Comment (lovio) [reply source/generator_cpp.inl:568]: https://github.com/stdbio/FBE/pull/25/files/ca7aff82b3e13a1cc534ab5137229d633c7a9e01#diff-f0615dde2c54edf88930da15bb35033bc24bd24b6decab5c9a1f1ab18bb0065bR561

File: proto/fbe.h (hunk 1 above)
Comment (lovio) [reply proto/fbe.h]: done

**Review by hurricane1026 — APPROVED** [review]

**Pull request merged**